snippets/typescript.json

```diff
@@ -0,0 +1,300 @@
+            "\t\taddressLine1: 'test',\n\t\tcity: 'test',\n\t}",
+            "\tunroll('it should #expectedStatus when #description', (",
+            "\t\t\tdone: () => void,",
+            "\t\t\targs: {id: number, inputBody: IUpdate${TM_FILENAME/(.*)\\.test.ts/${1}/}Body, expectedStatus: number, description: string}",
+            "\t): void => {\n\t\tupdate${TM_FILENAME/(.*)\\.test.ts/${1}/}(",
+            "\t\t\t{\n\t\t\t\tbody: JSON.stringify(args.inputBody),",
+            "\t\t\t\tpathParameters: {id: args.id},\n\t\t\t},",
+            "\t\t\tmockContext,\n\t\t\t(err: Error, response: any) => {",
+            "\t\t\t\texpect(response.statusCode).toEqual(args.expectedStatus);",
+            "\t\t\t}\n\t\t)\n\t\tdone();\n\t}, [",
+            "\t\t['id', 'inputBody', 'expectedStatus', 'description'],",
+            "\t\t[1, dummy${TM_FILENAME/(.*)\\.test.ts/${1}/}Data, 200, 'valid id is passed with valid data'],",
+            "\t\t[-1, dummy${TM_FILENAME/(.*)\\.test.ts/${1}/}Data, 500, 'invalid id is passed with valid data'],",
+            "\t\t[undefined, dummy${TM_FILENAME/(.*)\\.test.ts/${1}/}Data, 400, 'no id is passed'],",
+            "\t\t[1, null, 400, 'no data is passed'],\n\t])\n});\n"
+        ],
+        "description": "Snippet for testing Update Endpoint Lambda"
+    },
+
+    "Test for Delete Endpoint Lambda": {
+        "prefix": "testdelete",
+        "body": [
+            "describe('Test for delete${TM_FILENAME/(.*)\\.test.ts/${1}/}', (): void => {",
+            "\tunroll('it should #expectedStatus when #description', (\n\t\t\tdone: () => void,",
+            "\t\t\targs: {id: number, expectedStatus: number, description: string}",
+            "\t): void => {\n\t\tdelete${TM_FILENAME/(.*)\\.test.ts/${1}/}(",
+            "\t\t\t{\n\t\t\t\tpathParameters: {id: args.id},",
+            "\t\t\t},\n\t\t\tmockContext,",
+            "\t\t\t(err: Error, response: any) => {",
+            "\t\t\t\texpect(response.statusCode).toEqual(args.expectedStatus);",
+            "\t\t\t}\n\t\t)\n\t\tdone();\n\t}, [",
+            "\t\t['id', 'expectedStatus', 'description'],",
+            "\t\t[1, 200, 'valid data is passed'],",
+            "\t\t[-1, 500, 'invalid id is passed'],",
+            "\t\t[100, 500, 'non existent id is passed'],",
+            "\t\t[undefined, 400, 'no id is passed'],\n\t])\n});\n"
+        ],
+        "description": "Snippet for testing Delete Endpoint Lambda"
+    },
+
+    "Unroll for jest": {
+        "prefix": "unroll",
+        "body": [
+            "unroll('it should ${1:test message here}', (",
+            "\t\tdone: () => void,",
+            "\t\targs: {${2:argument: type}\\}",
+            "): void => {",
+            "\t${3}",
+            "\tdone();",
+            "}, [ // tslint:disable-next-line",
+            "\t[${0}],",
+            "\t[],",
+            "\t[],",
+            "]);\n"
+        ],
+        "description": "Snippet for unroll (testing)"
+    },
+
+    "it for jest": {
```

Comment:
`Test case for Jest`. Also this is present in the React extension. Do we need it here

Reply:
Yes we do, as the three extensions [`snips-react`, `snips-sequelize` and `snips-serverless`] are meant to be maintained separately, it only makes sense to keep the common snippets in all of them as one working on **react** (using `snips-react`) should not install `snips-sequelize` extension just to get the testing snippets. Leaving this as-is.